ExadelTimeTrackingSystem/ExadelTimeTrackingSystem.WebAPI/Infrastructure/ModelValidationActionFilter.cs

```diff
@@ -0,0 +1,75 @@
+﻿namespace ExadelTimeTrackingSystem.WebAPI.Filters
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using FluentValidation;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Mvc.Filters;
+
+    public class ModelValidationActionFilter : IAsyncActionFilter
+    {
+        private readonly IValidatorFactory _validatorFactory;
+
+        public ModelValidationActionFilter(IValidatorFactory validatorFactory)
+        {
+            _validatorFactory = validatorFactory;
+        }
+
+        public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+        {
+            var allErrors = new Dictionary<string, object>();
+
+            // Short-circuit if there's nothing to validate
+            if (context.ActionArguments.Count == 0)
+            {
+                await next();
+                return;
+            }
+
+            foreach (var (key, value) in context.ActionArguments)
```

Comment:
try to rename key and value to some meaningful names so it will be clear what they are used for